File: w06/d04/Liz_Goldstein/hangman.js
function Hangman(word) {

var word = chosenWord()
var guess_counter = 0;
var guessed_letters = []

function chosenWord() {
   return word[Math.floor(Math.random() * word.length)];
}

this.progress = function(){
    var playerProgress = [];
    for (var i = 0; i < word.length; i++) {
      playerProgress.push("_");
    }
    return playerProgress;
  };

this.guess = function(letter) {

    if (guess_counter === 0) {
      current_progress = this.progress();}
      var word_array = word.split("");
      for (var i = 0; i < word.length; i++) {
        if (word_array[i] === letter) {
          current_progress[i] = letter;}
          }

guessed_letters.push(letter)
guess_counter++

function gameStatus()
{ if (guess_counter === 5)
  {game_status =  "That was your last guess, game over";}
  else {game_status = "Keep trying" + " " + (5 - guess_counter) + " tries left. Letters you've guessed: " + guessed_letters;}
  return game_status;
}

if (guess_counter < 6)
{return current_progress + " " + gameStatus();}
else {return "NO YOUR GAME IS OVER SORRY";}
}

this.giveUp = function() {
  guess_counter = 6
  return "The word was" + " " + word + ". You gave up, please play again.";
}

this.seeProgress = function () {
  return "So far you've got: " + current_progress + " with " + (5 - guess_counter) + " left to guess. You've already guessed: " + guessed_letters ;
}

}

var game = new Hangman (["detective", "kitten", "puppy"]);
// console.log(game.progress());
console.log(game.guess("x"));
console.log(game.guess("t"));
console.log(game.guess("e"));
console.log(game.seeProgress());
console.log(game.giveUp());
console.log(game.guess("f"));
console.log(game.guess("l"));
console.log(game.guess("c"));

File: w10/d01/Victor_Rodriguez/hangman/lib/hangman.rb
class Hangman

  def initialize(word_array, _nums=Random.new) 
    word = word_array[_nums.rand(word_array.length)]
    @word = word.split("")
    @progress = @word.map { "_" }
    @wrong_guesses = 0
  end

  def progress()
    @progress.join(" ")
  end

  def guess?(guessed_letter)
    if @wrong_guesses < 5
      if @word.include?(guessed_letter)
        @word.each_with_index do |letter, index|
          if letter == guessed_letter
            @progress[index] = letter
          end
        end
        return true
      else
        @wrong_guesses += 1
        return false
      end
    else
      return nil
    end
  end
 
  def giveup()
    @wrong_guesses = 5
    return @word.join("")
  end

  
end